proto/batch_plan.proto

```diff
@@ -191,6 +191,17 @@ message MergeSortExchangeNode {
   repeated plan_common.ColumnOrder column_orders = 2;
 }
 
+message LookupJoinNode {
+  plan_common.JoinType join_type = 1;
+  expr.ExprNode condition = 2;
+  repeated int32 build_side_key = 3;
+  plan_common.CellBasedTableDesc probe_side_table_desc = 4;
+  repeated int32 probe_side_column_ids = 5;
+  repeated plan_common.Field probe_side_schema = 6;
+  repeated uint32 output_indices = 7;
+  repeated ExchangeSource sources = 8;
```

Comment:
Looking back at it, I think the first three can be constructed dynamically, however I don't see a way to avoid including `sources` because the only way `LookupJoinExecutor` can dynamically create sources is if it's given some `ExchangeSource` templates it can use to clone data from.